utilities/src/main/java/org/craftercms/commons/file/blob/impl/BlobStoreResolverImpl.java

```diff
@@ -0,0 +1,115 @@
+import java.util.function.Predicate;
+
+import static org.craftercms.commons.file.blob.BlobStore.CONFIG_KEY_ID;
+
+/**
+ * Default implementation of {@link BlobStoreResolver}
+ *
+ * @author joseross
+ * @since 3.1.6
+ */
+@SuppressWarnings("rawtypes, unchecked")
+public class BlobStoreResolverImpl implements BlobStoreResolver, ApplicationContextAware {
+
+    String CONFIG_KEY_STORE = "blobStore";
+
+    String CONFIG_KEY_TYPE = "type";
+
+    protected final Logger logger = LoggerFactory.getLogger(getClass());
+
+    /**
+     * The path of the configuration file
+     */
+    protected String configurationPath;
+
+    protected EncryptionAwareConfigurationReader configurationReader;
+
+    protected ApplicationContext applicationContext;
+
+    @Override
+    public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+        this.applicationContext = applicationContext;
+    }
+
+    public void setConfigurationPath(String configurationPath) {
+        this.configurationPath = configurationPath;
+    }
+
+    public void setConfigurationReader(EncryptionAwareConfigurationReader configurationReader) {
+        this.configurationReader = configurationReader;
+    }
+
+    protected BlobStore findStore(HierarchicalConfiguration config, Predicate<HierarchicalConfiguration> predicate)
+            throws ConfigurationException {
+        Optional<HierarchicalConfiguration> storeConfig =
+                config.configurationsAt(CONFIG_KEY_STORE).stream().filter(predicate).findFirst();
+        if (storeConfig.isPresent()) {
+            HierarchicalConfiguration store = storeConfig.get();
+            String type = store.getString(CONFIG_KEY_TYPE);
+            BlobStore instance = applicationContext.getBean(type, BlobStore.class);
+            instance.init(store);
+            return instance;
+        }
+        return null;
+    }
+
+    protected HierarchicalConfiguration getConfiguration(Function<String, InputStream> configGetter) {
+        logger.debug("Reading blob store configuration");
+        try (InputStream is = configGetter.apply(configurationPath)) {
+            if (is != null) {
+                return configurationReader.readXmlConfiguration(is, "utf-8");
+            }
+        } catch (Exception e) {
+            logger.error("Error reading blob store configuration", e);
+        }
+        return null;
+    }
+
+    @Override
+    public BlobStore getById(Function<String, InputStream> configGetter, String storeId) {
+        logger.debug("Looking blob store with id {}", storeId);
+        try {
+            HierarchicalConfiguration config = getConfiguration(configGetter);
+            if (config != null) {
+                return findStore(config, store ->
+                        StringUtils.equals(storeId, store.getString(CONFIG_KEY_ID)));
+            }
+        } catch (ConfigurationException e) {
+            logger.error("Error reading blob store configuration", e);
```

Comment:
Why are we swallowing the exception? Isn't this something the caller should know?